deploy/stacks/backend_stack.py

```diff
@@ -361,10 +361,10 @@ def __init__(
                 internet_facing=internet_facing,
             )
 
-    @run_if(["core.share_notifications.email.active"])
-    def create_ses_stack(self, custom_domain, envname, kwargs, resource_prefix, email_notification_sender_email_id):
-        if custom_domain == None and email_notification_sender_email_id == None and deploy_config.get_property("core.share_notifications.email.active") == True:
-            raise Exception("Cannot Create SES Stack For email notification as Custom Domain and Sender Email Id are not present. Either Disable Email Notification Config or add Custom Domain")
+    @run_if(["modules.datasets.share_notifications.email.active"])
+    def create_ses_stack(self, custom_domain, envname, kwargs, resource_prefix):
+        if custom_domain == None:
```

Comment:
UPDATE - looking at the code and the way we create `ses.EmailIdentity()` we are using the custom domain hosted zone create

I think to keep things simple for the initial version of this feature maybe we also use the custom_domain hosted zone (which will be required for email notifications) and we can remove the `email_sender_id` parameter from `cdk.json` -  @dlpzx @TejasRGitHub let me know thoughts

Reply:
I think just checking custom_domain to None would do. Initially , I was checking the custom_domain, email_notification_sender_email_id and the config for email notification. 

1. The config for email notification is now checked in the `run_if` decorator 
2. If the `email_notification_sender_email_id` is not present and email notification config is true then we can still use noreply@custom_domain as a default email-id .
3. If the custom_domain is not present ( and even is the email notification config is true and also the `email_notification_sender_email_id` is presnet / absent) then only we don't want the SES stack to be formed.

Reply:
@noah-paige, @TejasRGitHub 
- good point on checking the hosted zone of the `custom_domain` parameter, totally up for that. 
- When doing the initial reviews I think both of us got it wrong and the `email_sender_id` is not a required parameter, but an optional overwrite. If the parameter is going to be used (question for @TejasRGitHub) I say we should keep it. Now, where should we keep it? in cdk.json or config.json? For me the best options are either as 1. an additional subparameter in the `custom_domain` section of `cdk.json` or 2. as a feature of `share_notifications` in `config.json` or 3. as a feature of `core` in `config.json`

Reply:
Hi @dlpzx , yes you are right in which the `email_sender_id` is not a required parameter. I had added that thinking if any one might want to have some other sender email id. But I am open to not having it in the first place in the `cdk.json`. If we remove it the sender email id will always be noreply@`custom_domain`. 

If we are keeping it, having the `email_sender_id ` in the `custom_domain`, doesn't make sense as it is not a domain related thing. If we want to move it out of `cdk.json`, I think probably in the `share_notifications` in `config.json`.

Reply:
It is commented below, but the decision made was to keep the `email_notification_sender_email_id` as part of the custom_domain block as it can only exist when the custom domain is provided
```
"custom_domain": {
          "hosted_zone_name": "string_ROUTE_53_EXISTING_DOMAIN_NAME|DEFAULT=None, REQUIRED if internet_facing=false",
          "hosted_zone_id": "string_ROUTE_53_EXISTING_HOSTED_ZONE_ID|DEFAULT=None, REQUIRED if internet_facing=false",
          "certificate_arn": "string_AWS_CERTIFICATE_MANAGER_ARN|DEFAULT=None, REQUIRED if internet_facing=false",
          "email_notification_sender_email_id": "string_EMAIL_NOTIFICATION_SENDER_EMAIL_ID|DEFAULT=None"
        },
```